### PR TITLE
Control lora trainer hunyuan

### PR DESCRIPTION
**Pull request by neph1**
Still have some cleaning up to do, but this pr is back (I might migrate this and force push, or if it's eventually squashed).
It does things and produced something. I will do a longer run tomorrow, but I have no idea how to inference. ~Is there an example script somewhere i can modify?~ Found it

### CODE REVIEW

File: finetrainers/models/hunyuan_video/base_specification.py [hunk 1]
```diff
@@ -57,19 +59,25 @@ def forward(
         assert video.ndim == 5, f"Expected 5D tensor, got {video.ndim}D tensor"
         video = video.to(device=device, dtype=vae.dtype)
         video = video.permute(0, 2, 1, 3, 4).contiguous()  # [B, F, C, H, W] -> [B, C, F, H, W]
-
+        
+        compute_posterior = False
```
Comment (neph1) [inline finetrainers/models/hunyuan_video/base_specification.py:63]: So far only made it work with compute_posterior false

**Comment by neph1** [comment]
I will try a first frame conditioning run, since I believe using the hunyuan i2v pipeline would be able to verify if it works or not. My current test run was for canny using hunyuan t2v, which was probably a mistake.

**Review by a-r-r-o-w — COMMENTED** [review]
Hey @neph1, thanks for the PR and testing with some runs! I know there's some duplication of code at the moment, but I plan to address that in the future with something else. For now, let's keep the duplication

I'll try to help with launching some training runs too for verifying the PR once we have the changes looking similar to the Wan/CogView control implementations #338

**Comment by neph1** [comment]
Oh, I did follow the new implementation, just kept it on the same branch. At least I believe it follows what is in main now.
Not sure it works, though. I ran 1000 steps canny training on some simpsons dataset I found, think I managed to make the conversion script handle x_embedder, but it's only noise. Sadly can't inference with plain diffusers due to VRAM.

Got a bit disheartened, but I'll get back to it.

File: finetrainers/models/hunyuan_video/base_specification.py [hunk 2]
```diff
@@ -58,18 +58,24 @@ def forward(
...
+            # if vae.use_slicing and video.shape[0] > 1:
+            #     encoded_slices = [vae._encode(x_slice) for x_slice in video.split(1)]
+            #     moments = torch.cat(encoded_slices)
+            # else:
+            #     moments = vae._encode(video)
+            moments = vae._encode(video)
             latents = moments.to(dtype=dtype)
 
-        return {self.output_names[0]: latents}
+        latents_mean = torch.tensor(vae.latent_channels)
```
Comment (a-r-r-o-w) [inline finetrainers/models/hunyuan_video/base_specification.py:75]: @neph1 These changes seem incorrec to me and will cause worse generations. The previous implementation that did not perform this normalization was correct, I think.

Was this modified from Wan? If so, it's incorrect because they are different models and preprocess latents differently

**Comment by neph1** [comment]
By all means, if you have the time. :+1:
